server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
Parameters themselves aren't covarient, you'll end up overloading instead of overriding. So, to get type-specific, overrideable methods, those interfaces and types were added.